client/src/components/CollegeShortlistModal.js

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import {Button, Modal, Grid, Label, Icon, Dropdown } from 'semantic-ui-react';
+import PooledSingleSelectDropdown from "./PooledSingleSelectDropdown"
+import { makeCall } from "../apis";
+import swal from "sweetalert";
+
+/*
+    Modal makes a PATCH call to update alumni information
+    props:
+    - modalOpen
+    - closeModal
+    - getInput
+    - id
+*/
+export default class CollegeShortlistModal extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            existingCollegeId: '', // mongoId when existing college selected
+            existingCollegeName: '',
+            countryOptions: [],
+            submitting: false,
+        }
+        this.handleCountrySelection = this.handleCountrySelection.bind(this)
+        this.getCollegeInput = this.getCollegeInput.bind(this)
+        this.deleteCountry = this.deleteCountry.bind(this)
+        this.submit = this.submit.bind(this)
+        this.clearState = this.clearState.bind(this)
+    }
+
+    async componentWillMount() {
+        let result = await makeCall(null, '/drop/countries', 'get')
+        this.setState({countryOptions: result.options})
+    }
+
+    handleCountrySelection(e, {value}) {
+        e.preventDefault()
+        this.setState({
+            country: value
+        })
+    }
+
+    // getCollegeInput(selection, isNew) {
+    getCollegeInput(selection) {
+        if (selection) {
+            this.setState({
+                existingCollegeId: selection.value,
+                existingCollegeName: selection.text
+            })
+        } else {
+            this.setState({
+                existingCollegeId: '',
+                existingCollegeName: ''
+            })
+        }
+    }
+
+    deleteCountry(e) {
+        e.preventDefault()
+        this.setState({
+            country: '',
+            newCollege: '',
+            existingCollegeId: '',
+            existingCollegeName: ''
+        })
+    }
+
+    clearState() {
+        this.setState({
+            country: '',
+            existingCollegeId: '',
+            existingCollegeName: ''
+        })
+    }
+
+    findDuplicate = (newCollege) => {
```

Comment:
nit: since this returns a boolean can we name the method `isDuplicateCollegeEntry`